serde/src/de/impls.rs

```diff
@@ -180,6 +180,28 @@ macro_rules! num_as_self {
     };
 }
 
+macro_rules! num_as_copysign_self {
+    ($ty:ident : $visit:ident) => {
+        #[inline]
+        fn $visit<E>(self, v: $ty) -> Result<Self::Value, E>
+        where
+            E: Error,
+        {
+            #[cfg(any(no_float_copysign, not(feature = "std")))]
+            {
+                Ok(v as Self::Value)
+            }
```

Comment:
This produces nondeterministic wrong values if serde is built with "std" feature disabled. https://github.com/rust-lang/rust/issues/50145

I'm not sure whether we can manually implement a correct copysign without going through the LLVM copysign intrinsics. If that were possible, I presume libcore would already have done it?